### PR TITLE
add API for bug reporting

### PR DESCRIPTION
**Pull request by kevinushey**
### Intent

This PR adds two functions, `.rs.bugReport()` and `.rs.api.bugReport()`, intended to be used both internally by us and by users to report bugs in the IDE. The goal is to automate some of the R + RStudio session information discovery, so bug reporters don't need to manually update the issue template for new bugs.

The internal version `.rs.bugReport()` is intended to be used by us, and can be used to report product bugs in both the open source and the Pro products.

Currently, the API function is only used for filing bugs on the open-source repository.

Addresses https://github.com/rstudio/rstudio/issues/10221.

### Approach

> Describe the approach taken and the tradeoffs involved if non-obvious; add an overview of the solution if it's complicated.

### Automated Tests

N/A

### QA Notes

Test that you can create a bug report using `.rs.bugReport()`.

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

**Comment by kevinushey** [comment]
Here's an example of what the generated template looks like:

---

### System details

RStudio Edition : 2022.1.0.303
RStudio Version : Desktop [Open Source]
OS Version      : macOS Monterey 12.0.1
R Version       : R version 4.1.2 (2021-11-01)

### Steps to reproduce the problem

### Describe the problem in detail

### Describe the behavior you expected

---

<details>

<summary>Session Info</summary>

``` r
R version 4.1.2 (2021-11-01)
Platform: aarch64-apple-darwin20.6.0 (64-bit)
Running under: macOS Monterey 12.0.1

Matrix products: default
LAPACK: /opt/homebrew/Cellar/r/4.1.2/lib/R/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] shiny_1.7.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.7        bslib_0.3.1       compiler_4.1.2    pillar_1.6.4      later_1.3.0       jquerylib_0.1.4   highr_0.9         tools_4.1.2       digest_0.6.29     jsonlite_1.7.2   
[11] evaluate_0.14     lifecycle_1.0.1   tibble_3.1.6      pkgconfig_2.0.3   rlang_0.4.12      reprex_2.0.1.9000 cli_3.1.0         rstudioapi_0.13   yaml_2.2.1        xfun_0.28        
[21] fastmap_1.1.0     withr_2.4.3       knitr_1.36        fs_1.5.2          sass_0.4.0        vctrs_0.3.8       grid_4.1.2        glue_1.5.1        R6_2.5.1          processx_3.5.2   
[31] fansi_0.5.0       rmarkdown_2.11    callr_3.7.0       clipr_0.7.1       magrittr_2.0.1    ps_1.6.0          promises_1.2.0.1  ellipsis_0.3.2    htmltools_0.5.2   mime_0.12        
[41] xtable_1.8-4      httpuv_1.6.3      utf8_1.2.2        miniUI_0.1.1.1    cachem_1.0.6      crayon_1.4.2     
```

</details>

File: src/cpp/r/R/Tools.R [hunk 1]
```diff
@@ -1290,6 +1290,113 @@ environment(.rs.Env[[".rs.addFunction"]]) <- .rs.Env
...
+      op <- options(width = 78)
+      on.exit(options(op), add = TRUE)
+      paste(capture.output(print(rInfo)), collapse = "\n")
+   })
+   
+   # create issue template and fill in the pieces
+   template <- .rs.heredoc("
+      ### System details
+
+      RStudio Edition : %s
```
Comment (jmcphers) [inline src/cpp/r/R/Tools.R:1334]: Nit - doesn't look like these lines are going to get the monospace treatment so the `:` will get staggered. Maybe change the formatting so they get monospaced like our real bug template, or just remove all the spacing so it doesn't look like they're aligned?

**Review by jmcphers — APPROVED** [review]
Neat idea!

File: src/cpp/r/R/Tools.R (hunk 1 above)
Comment (kevinushey) [reply src/cpp/r/R/Tools.R:1334]: Good point; I'll make that change. Thanks!

**Pull request merged**